### ISSUE TITLE
conv2d boundaries

### ISSUE THREAD

**Issue by kercos** [pytorch/pytorch#137086]
### 🚀 The feature, motivation and pitch

[torch.nn.functional.conv2d](https://pytorch.org/docs/stable/generated/torch.nn.functional.conv2d.html) (and respective [torch.nn.Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html)) miss `boundaries` which is implemented on scipy [convolve2d](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.convolve2d.html).

Any plans to implement this feature?

**Comment by albanD** [comment]
Hi, this concept is called "padding" in the context of PyTorch.
In particular, you can see on Conv2d the option: padding_mode (str, optional) – 'zeros', 'reflect', 'replicate' or 'circular'. Default: 'zeros'

**Comment by kercos** [comment]
Hi @albanD, thanks for pointing this out! `padding_mode` is indeed what I was looking for. I noticed this is only in the class implementation and not in the functional implementation. 
- [v. 2.4 functional conv2d](https://pytorch.org/docs/2.4/generated/torch.nn.functional.conv2d.html): NO `padding_mode`
- [v. 2.4 class conv2d](https://pytorch.org/docs/2.4/generated/torch.nn.Conv2d.html#torch.nn.Conv2d): YES `padding_mode`

Any specific reason why so?

**Comment by albanD** [comment]
This is because the module handles these complex padding at a higher level here: https://github.com/pytorch/pytorch/blob/3c1ab9367885fdb0ead5fcc14a22d6934070ca92/torch/nn/modules/conv.py#L539-L541

We could add this to the functional version but that would require adding these (relatively complex) padding techniques to the c++ implementation directly.
